File: app/src/main/java/com/example/studybuddy/YourFlashcardsView.java
package com.example.studybuddy;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import kotlin.Metadata;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;


public final class YourFlashcardsView extends AppCompatActivity {
    TextView cardContent;
    ArrayList<FlashcardModel> flashcards;
    ImageButton rightArrow;
    ImageButton leftArrow;
    TextView setName;
    TextView progress;
    FirebaseAuth mAuth;
    boolean showTerm;
    int i;
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_your_flashcards_view);
        i = 0;
        showTerm = true;
        cardContent = findViewById(R.id.cardcontent);
        leftArrow = findViewById(R.id.leftarrow);
        rightArrow = findViewById(R.id.rightarrow);
        setName = findViewById(R.id.viewSetName);
        flashcards = new ArrayList<FlashcardModel>();
        progress = findViewById(R.id.progress);
        mAuth = FirebaseAuth.getInstance();
        Intent intent = getIntent();
        String name = intent.getStringExtra("name");
        String id = intent.getStringExtra("id");
        setName.setText(name);


        //Replace with data later

        DatabaseReference ref = FirebaseDatabase.getInstance().getReference().child(mAuth.getCurrentUser().getEmail().split("@")[0]).child(id).child("flashcards");
        ref.addValueEventListener(new ValueEventListener() {
                                      @Override
                                      public void onDataChange(DataSnapshot dataSnapshot) {
                                          for (DataSnapshot item_snapshot : dataSnapshot.getChildren()) {
                                              String term = item_snapshot.child("definition").getValue().toString();
                                              String definition = item_snapshot.child("term").getValue().toString();
                                              flashcards.add(new FlashcardModel(term, definition));
                                          }
                                          cardContent.setText(flashcards.get(0).term);
                                          progress.setText(1 + "/" + flashcards.size());
                                      }


                                      @Override
                                      public void onCancelled(@NonNull DatabaseError error) {
                                      }
                                  });

        rightArrow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                i++; i%=flashcards.size();
                cardContent.setText(flashcards.get(i).term);
                progress.setText(i+1 + "/" + flashcards.size());
            }
        });


        leftArrow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                i--;
                if (i == -1) {
                    i = flashcards.size() - 1;
                }
                progress.setText(i+1 + "/" + flashcards.size());
                cardContent.setText(flashcards.get(i).term);

            }
        });

        cardContent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showTerm = !showTerm;
                if (showTerm==false) {
                    cardContent.setText(flashcards.get(i).definition);
                } else {
                    cardContent.setText(flashcards.get(i).term);
                }
            }
        });
    }
}

File: app/src/main/java/com/example/studybuddy/YourFlashcardsView.kt
package com.example.studybuddy

import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import com.example.studybuddy.FlashcardModel
import android.widget.ImageButton
import com.google.firebase.auth.FirebaseAuth
import android.os.Bundle
import com.example.studybuddy.R
import android.content.Intent
import android.view.View
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import java.util.ArrayList

class YourFlashcardsView : AppCompatActivity() {
    lateinit var cardContent: TextView
    lateinit var flashcards: ArrayList<FlashcardModel>
    lateinit var rightArrow: ImageButton
    lateinit var leftArrow: ImageButton
    lateinit var setName: TextView
    lateinit var progress: TextView
    lateinit var mAuth: FirebaseAuth
    var showTerm = false
    var i = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.setContentView(R.layout.activity_your_flashcards_view)
        i = 0
        showTerm = true
        cardContent = findViewById(R.id.cardcontent)
        leftArrow = findViewById(R.id.leftarrow)
        rightArrow = findViewById(R.id.rightarrow)
        setName = findViewById(R.id.viewSetName)
        flashcards = ArrayList()
        progress = findViewById(R.id.progress)
        mAuth = FirebaseAuth.getInstance()
        val intent = intent
        val name = intent.getStringExtra("name")
        val id = intent.getStringExtra("id")
        setName.setText(name)


        //Replace with data later
        val ref = FirebaseDatabase.getInstance().reference.child(
            mAuth!!.currentUser!!.email!!
                .split("@").toTypedArray()[0]
        ).child(id!!).child("flashcards")
        ref.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                for (item_snapshot in dataSnapshot.children) {
                    val term = item_snapshot.child("term").value.toString()
                    val definition = item_snapshot.child("definition").value.toString()
                    flashcards!!.add(FlashcardModel(term, definition))
                }
                cardContent.setText(flashcards!![0].term)
                progress.setText(1.toString() + "/" + flashcards!!.size)
            }

            override fun onCancelled(error: DatabaseError) {}
        })
        rightArrow.setOnClickListener(View.OnClickListener {
            i++
            i %= flashcards!!.size
            cardContent.setText(flashcards!![i].term)
            progress.setText(i.toString() + 1.toString() + "/" + flashcards!!.size)
        })
        leftArrow.setOnClickListener(View.OnClickListener {
            i--
            if (i == -1) {
                i = flashcards!!.size - 1
            }
            progress.setText(i.toString() + 1.toString() + "/" + flashcards!!.size)
            cardContent.setText(flashcards!![i].term)
        })
        cardContent.setOnClickListener(View.OnClickListener {
            showTerm = !showTerm
            if (showTerm == false) {
                cardContent.setText(flashcards!![i].definition)
            } else {
                cardContent.setText(flashcards!![i].term)
            }
        })
    }
}